syntax = "proto3";

package com.eyeo.ctu.engine.pb;
option java_package = "com.eyeo.ctu.engine.pb.wire";
option java_multiple_files = true;
// optimize_for = LITE_RUNTIME is deprecated and thus not used here
// (https://github.com/protocolbuffers/protobuf/blob/v3.9.0/java/lite.md)

enum ContentType {
    Image = 0;
    Script = 1;
    SubDocument = 2;
}

message MatchesRequest {
    string url = 1;
    repeated ContentType contentTypes = 2;
    repeated string documentUrls = 3;
    string sitekey = 4;
    bool specificOnly = 5;
}

message BlockingFilter {
    uint64 pointer = 1;
}

message MatchesResponse {
    BlockingFilter filter = 1;
}